gossip/src/cluster_info.rs

```diff
@@ -990,7 +990,7 @@ impl ClusterInfo {
         }
     }
 
-    pub fn push_vote(&self, tower: &[Slot], vote: Transaction) {
+    pub fn push_vote(&self, tower: &[Slot], vote: Transaction, vote_slot: Slot) {
```

Comment:
Do we actually need `vote_slot: Slot`?
Isn't that the same as `tower.last()`?
We can add debug asserts that these are the same.

Reply:
I think we don't want to rely on the assumption that tower is sorted here.

Reply:
We actually have a debug check for that:
https://github.com/solana-labs/solana/blob/e3915e4b7/gossip/src/cluster_info.rs#L994

The more invariant we enforce in the code, the lesser likely would be to add bugs.

Reply:
ah I didn't see that, makes sense